aspnetcore-webapi/GettingStartedASPNETMongoDB/GettingStartedASPNETMongoDB/Controllers/StudentsController.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using GettingStartedASPNETMongoDB.Interfaces;
+using GettingStartedASPNETMongoDB.Models;
+
+namespace GettingStartedASPNETMongoDB.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class StudentsController : ControllerBase
+    {
+        private readonly IStudentService _studentService;
+
+        private readonly ICourseService _courseService;
+
+        public StudentsController(IStudentService studentService, ICourseService courseService)
+        {
+            _studentService = studentService;
+
+            _courseService = courseService;
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Student>>> GetAll()
+        => Ok(await _studentService.GetAll());
+
+        [HttpGet("{id:length(24)}")]
+        public async Task<ActionResult<Student>> GetById(string id)
+        {
+            var student = await _studentService.GetById(id);
+
+            if (student is null)
+            {
+                return NotFound();
+            }
+                
+            if (student.Courses?.Count > 0)
+            {
+                var courseList = new List<Course>();
+
+                foreach (var course in student.Courses)
+                {
+                    var studentCourse = await _courseService.GetById(course);
+
+                    if (studentCourse != null)
+                    {
+                        courseList.Add(studentCourse);
+                    }
+                }
+
+                student.CourseList = courseList;
+            }
```

Comment:
Here we don't need to check if the courses count is more than 0, we can just proceed with the execution of the logic, and it's also a good practice, to initialize the list to an empty list even if there are no courses, an empty list is better than null in this context. And we won't be needing an extra variable `var courseList` you can use `student.CourseList` directly.